packages/hash/api/src/model/knowledge/workspaceInstance.model.ts

```diff
@@ -0,0 +1,180 @@
+import { GraphApi } from "../../graph";
+import {
+  WorkspaceInstanceModel,
+  EntityModel,
+  EntityModelCreateParams,
+  UserModel,
+  LinkModel,
+} from "..";
+import { workspaceAccountId } from "../util";
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+import { EntityTypeMismatchError, NotFoundError } from "../../lib/error";
+
+export type WorkspaceInstanceModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel" | "ownedById"
+> & {};
+
+/**
+ * @class {@link WorkspaceInstanceModel}
+ */
+export default class extends EntityModel {
+  static fromEntityModel(entity: EntityModel): WorkspaceInstanceModel {
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.workspaceInstance.schema.$id
+    ) {
+      throw new EntityTypeMismatchError(
+        entity.entityId,
+        WORKSPACE_TYPES.entityType.workspaceInstance.schema.$id,
+        entity.entityTypeModel.schema.$id,
+      );
+    }
+
+    return new WorkspaceInstanceModel(entity);
+  }
+
+  /**
+   * Create the workspace instance entity.
+   *
+   * @see {@link EntityModel.create} for the remaining params
+   */
+  static async createWorkspaceInstance(
+    graphApi: GraphApi,
+    params: WorkspaceInstanceModelCreateParams,
+  ) {
+    // Ensure the workspace instance entity has not already been created.
+    await WorkspaceInstanceModel.getWorkspaceInstanceModel(graphApi)
+      .then(() => {
+        throw new Error("Workspace instance entity already exists.");
+      })
+      .catch((error: Error) => {
+        if (!(error instanceof NotFoundError)) {
+          throw error;
+        }
```

Comment:
This error setup is a little strange to me. Could we get away with just the catch? I would assume the `getWorkspaceInstanceModel` shouldn't be errorful in the `then`

Reply:
> Could we get away with just the catch? I would assume the getWorkspaceInstanceModel shouldn't be errorful in the then

If a HASH instance entity exists, I need to throw so that a second HASH instance entity isn't created. That's why I am throwing in the `.then`. Would you prefer something like this instead?

```TypeScript
    const existingHashInstance = await HashInstanceModel.getHashInstanceModel(
      graphApi,
    ).catch((error: Error) => {
      if (error instanceof NotFoundError) {
        return null;
      }
      throw error;
    });

    if (existingHashInstance) {
      throw new Error("Hash instance entity already exists.");
    }
```

Reply:
I see. I misunderstood the comment above the code the first time around. But I think I would prefer something like what you suggest here, though. It might just be the sort of "negative" control-flow which makes it a little difficult to follow

Reply:
Addressed in https://github.com/hashintel/hash/pull/1327/commits/dcc57957649239f7bbc2496314e3001fab753400